From 0d0bd3f7c0c3860fbe02732f69b1800f3389d0a8 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 19 Jul 2022 14:49:57 +0800
Subject: [PATCH 40/62] Revert "drm/msm/dsi: fix error checks and return values
 for DSI xmit functions"

This reverts commit e99755e6a99293c7cb88a71ad2dc4e02eba6354b.
---
 drivers/gpu/drm/msm/dsi/dsi_host.c | 21 +++++++--------------
 1 file changed, 7 insertions(+), 14 deletions(-)

diff --git a/drivers/gpu/drm/msm/dsi/dsi_host.c b/drivers/gpu/drm/msm/dsi/dsi_host.c
index 51e8318cc8ff..64454a63bbac 100644
--- a/drivers/gpu/drm/msm/dsi/dsi_host.c
+++ b/drivers/gpu/drm/msm/dsi/dsi_host.c
@@ -1371,10 +1371,10 @@ static int dsi_cmds2buf_tx(struct msm_dsi_host *msm_host,
 			dsi_get_bpp(msm_host->format) / 8;
 
 	len = dsi_cmd_dma_add(msm_host, msg);
-	if (len < 0) {
+	if (!len) {
 		pr_err("%s: failed to add cmd type = 0x%x\n",
 			__func__,  msg->type);
-		return len;
+		return -EINVAL;
 	}
 
 	/* for video mode, do not send cmds more than
@@ -1393,14 +1393,10 @@ static int dsi_cmds2buf_tx(struct msm_dsi_host *msm_host,
 	}
 
 	ret = dsi_cmd_dma_tx(msm_host, len);
-	if (ret < 0) {
-		pr_err("%s: cmd dma tx failed, type=0x%x, data0=0x%x, len=%d, ret=%d\n",
-			__func__, msg->type, (*(u8 *)(msg->tx_buf)), len, ret);
-		return ret;
-	} else if (ret < len) {
-		pr_err("%s: cmd dma tx failed, type=0x%x, data0=0x%x, ret=%d len=%d\n",
-			__func__, msg->type, (*(u8 *)(msg->tx_buf)), ret, len);
-		return -EIO;
+	if (ret < len) {
+		pr_err("%s: cmd dma tx failed, type=0x%x, data0=0x%x, len=%d\n",
+			__func__, msg->type, (*(u8 *)(msg->tx_buf)), len);
+		return -ECOMM;
 	}
 
 	return len;
@@ -2143,12 +2139,9 @@ int msm_dsi_host_cmd_rx(struct mipi_dsi_host *host,
 		}
 
 		ret = dsi_cmds2buf_tx(msm_host, msg);
-		if (ret < 0) {
+		if (ret < msg->tx_len) {
 			pr_err("%s: Read cmd Tx failed, %d\n", __func__, ret);
 			return ret;
-		} else if (ret < msg->tx_len) {
-			pr_err("%s: Read cmd Tx failed, too short: %d\n", __func__, ret);
-			return -ECOMM;
 		}
 
 		/*
-- 
2.27.0

